contracts/paymaster/src/forward_call.rs

```diff
@@ -0,0 +1,62 @@
+multiversx_sc::imports!();
+
+pub type PaymentsVec<M> = ManagedVec<M, EsdtTokenPayment<M>>;
+
+static ERR_CALLBACK_MSG: &[u8] = b"Error received in callback:";
+pub const ESDT_TRANSFER_FUNC_NAME: &str = "ESDTTransfer";
+#[multiversx_sc::module]
+pub trait ForwardCall {
+    fn forward_call(
+        &self,
+        dest: ManagedAddress,
+        endpoint_name: ManagedBuffer,
+        endpoint_args: MultiValueEncoded<ManagedBuffer>,
+        payments: PaymentsVec<Self::Api>,
+    ) {
+        let original_caller = self.blockchain().get_caller();
+
+        let (contract_call_endpoint, contract_call_args) =
+            if !self.blockchain().is_smart_contract(&dest) {
```

Comment:
No need for this if. You can do simple transfer to users, but add all arguments received here as data. Or just delete the if statement and do asyncCall as that will resolve it.